eng/common/scripts/Download-Latest-Contract.ps1

```diff
@@ -0,0 +1,71 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory,
+    [ValidateNotNullOrEmpty()]
+    [String] $TagName

```

Comment:
Can we determine this from the tag name?  I think for each language it might be a different algo but we can use the delegate pattern [like this](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Cadl-Project-Generate.ps1#L57).